src/Proto.Remote.GrpcCore/Remote.cs

```diff
@@ -0,0 +1,113 @@
+            _config = config;
+            System.Extensions.Register(this);
+        }
+        public bool Started { get; private set; }
+        public ActorSystem System { get; }
+        public RemoteConfigBase Config => _config;
+
+        public Task StartAsync()
+        {
+            lock (this)
+            {
+                if (Started)
+                    return Task.CompletedTask;
+                var channelProvider = new ChannelProvider(_config);
+                _endpointManager = new EndpointManager(System, Config, channelProvider);
+                _endpointReader = new EndpointReader(System, _endpointManager, Config.Serialization);
+                _healthCheck = new HealthServiceImpl();
+                _server = new Server
+                {
+                    Services =
+                    {
+                        Remoting.BindService(_endpointReader),
+                        Health.BindService(_healthCheck)
+                    },
+                    Ports = { new ServerPort(Config.Host, Config.Port, _config.ServerCredentials) }
+                };
+                _server.Start();
+
+                var boundPort = _server.Ports.Single().BoundPort;
+                System.SetAddress(Config.AdvertisedHost ?? Config.Host, Config.AdvertisedPort ?? boundPort
+                );
+                _endpointManager.Start();
+
+                Logger.LogDebug("Starting Proto.Actor server on {Host}:{Port} ({Address})", Config.Host, boundPort,
+                    System.Address
+                );
+                Started = true;
+                return Task.CompletedTask;
+            }
+        }
+
+        public Task ShutdownAsync(bool graceful = true)
+        {
+            lock (this)
+            {
+                if (!Started)
+                    return Task.CompletedTask;
+                try
+                {
+                    if (graceful)
+                    {
+                        _endpointManager.Stop();
+                        _server.ShutdownAsync().GetAwaiter().GetResult();
+                    }
+                    else
+                    {
+                        _server.KillAsync().GetAwaiter().GetResult();
+                    }
+
+                    Logger.LogDebug(
+                        "Proto.Actor server stopped on {Address}. Graceful: {Graceful}",
+                        System.Address, graceful
+                    );
+                }
+                catch (Exception ex)
+                {
+                    Logger.LogError(
+                        ex, "Proto.Actor server stopped on {Address} with error: {Message}",
+                        System.Address, ex.Message
+                    );
+                    _server.KillAsync().GetAwaiter().GetResult();
+                }
+                Started = false;
+                return Task.CompletedTask;
+            }
+        }
+
+        public void SendMessage(PID pid, object msg, int serializerId)

```

Comment:
Is it only used in unit tests ?

Reply:
Might be after your changes, in the dev branch its used from a few more places

<img width="1065" alt="Skärmavbild 2020-11-13 kl  19 40 19" src="https://user-images.githubusercontent.com/647031/99108791-2480e700-25e8-11eb-9dde-27bb7cc84dfd.png">